airflow/api/common/experimental/delete_dag.py

```diff
@@ -42,7 +42,11 @@ def delete_dag(dag_id: str, keep_records_in_log: bool = True, session=None) -> i
     """
     log.info("Deleting DAG: %s", dag_id)
     running_tis = (
-        session.query(models.TaskInstance.state).filter(models.TaskInstance.state.in_(State.unfinished)).all()
+        session.query(models.TaskInstance.state)
+        .filter(models.TaskInstance.dag_id == dag_id)
+        .filter(models.TaskInstance.state == State.RUNNING)
+        .limit(1)
+        .count()
```

Comment:
Wait "limit 1" and count does not make sense to me

Reply:
```suggestion
        .first()
```

Reply:
https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.first

Reply:
And we have the same logic on L53 too so let's use that

Reply:
That was what we did initially. What do you think @uranusjr @ashb 

I agree, using limit and count as above doesn't sound good.

Reply:
Happy to change it, if requested